src/sentry/static/sentry/app/components/group/externalIssuesList.tsx

```diff
@@ -116,19 +108,43 @@ class ExternalIssueList extends AsyncComponent<Props, State> {
     }
   }
 
+  async updateIntegrations(onSuccess = () => {}, onError = () => {}) {
+    try {
+      const {group} = this.props;
+      let integrations = await this.api.requestPromise(
+        `/groups/${group.id}/integrations/`
+      );
+      this.setState({integrations}, () => onSuccess());
+    } catch (error) {
+      onError();
+    }
+  }
+
   renderIntegrationIssues(integrations: GroupIntegration[] = []) {
     const {group} = this.props;
 
     const activeIntegrations = integrations.filter(
       integration => integration.status === 'active'
     );
 
+    const activeIntegrationsByProvider: {
+      [key: string]: GroupIntegration[];
+    } = activeIntegrations.reduce((acc, curr) => {
+      if (acc[curr.provider.key]) {
+        acc[curr.provider.key].push(curr);
+      } else {
+        acc[curr.provider.key] = [curr];
+      }
+      return acc;
+    }, {});
+
     return activeIntegrations.length
-      ? activeIntegrations.map(integration => (
+      ? Object.entries(activeIntegrationsByProvider).map(([provider, configurations]) => (
```

Comment:
@NisanthanNanthakumar should we sort by the provider? Not sure if `Object.entries` will give us deterministic ordering each time.

Reply:
@scefali hmm `Object.entries` does not keep order, but `Map.entries` does. I'm going to convert it into a Map.